Include/object.h

```diff
@@ -279,13 +279,9 @@ _Py_ThreadId(void)
     __asm__ ("" : "=r" (tp));
     tid = tp;
     #endif
-#elif defined(__s390__)
-    #if defined(__clang__) && _Py__has_builtin(__builtin_thread_pointer)
+#elif defined(__s390__) && _Py__has_builtin(__builtin_thread_pointer)
```

Comment:
I don't think we should guard this with `_Py__has_builtin()`. That's only supported in GCC 10+, but older versions of GCC support `__builtin_thread_pointer()`.

Something like:

`#elif defined(__s390__) && defined(__GNUC__)`

should work.

Reply:
Ah.. `__has_builtin` was added from gcc 10 :(